### PR TITLE
Add circleci to Tridactyl

### PR DESCRIPTION
**Pull request by glacambre**

### CODE REVIEW

**Comment by glacambre** [comment]
Shellcheck is disabled for now - I'll fix it in another PR

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+version: 2.1
+commands:
+  commoncmd:
+    steps:
+      - run: sudo apt update
+      - run: sudo apt install -qq shellcheck
+      - checkout
```
Comment (bovine3dom) [inline .circleci/config.yml]: Shellcheck doesn't look very removed :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          keys:
+            - dependency-cache-{{ checksum "package.json" }}
+      - run: npm install
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+      - run: npm run build
+jobs:
+  check:
```
Comment (bovine3dom) [inline .circleci/config.yml]:
```suggestion
  lint:
```

Will make it easier to work out what's going on.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    docker:
+      - image: circleci/node:latest
+    steps:
+      - commoncmd
+      # - run: bash -c 'GLOBIGNORE="node_modules" shellcheck **/*.sh'
+      - run: npm run lint
+      - run: bash -c '"$(npm bin)/tslint" --project .'
+      - run: bash -c '"$(npm bin)/jest" --silent src'
+  test:
```
Comment (bovine3dom) [inline .circleci/config.yml]: Perhaps call this unit so the e2e can be called e2e with ambiguity

**Pull request merged**